Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h

```diff
@@ -101,117 +101,190 @@
 #define TEMP_BED_PIN                        PC0   // TB1
 
 //
-// Heaters / Fans
+// Heaters
 //
 #define HEATER_0_PIN                        PC7   // HEATER1
 #define HEATER_1_PIN                        PA6   // HEATER2
 #define HEATER_BED_PIN                      PC6   // HOT BED
 
+//
+// Fan
+//
 #define FAN_PIN                             PA7   // FAN
 
-/**
- * Note: MKS Robin board is using SPI2 interface. Make sure your stm32duino library is configured accordingly
- */
+//
+// Thermocouples
+// 
 //#define MAX6675_SS_PIN                    PE5   // TC1 - CS1
 //#define MAX6675_SS_PIN                    PE6   // TC2 - CS2
 
+//
+// Filament runout sensor
+//
+#define FIL_RUNOUT_PIN                      PF11  // MT_DET
+
+//
+// Power loss detection
+//
 #define POWER_LOSS_PIN                      PA2   // PW_DET
+
+//
+// Power supply control
+//
 #define PS_ON_PIN                           PA3   // PW_OFF
-#define FIL_RUNOUT_PIN                      PF11  // MT_DET
 
-#ifdef ARDUINO_ARCH_STM32F1
-  #define BEEPER_PIN                        PC13
-#else
-  #define BEEPER_PIN                        -1
-#endif
+//
+// Piezzoelectric speaker
+//
+#define BEEPER_PIN                          PC13
```

Comment:
I wonder why this was `-1` when using the STM32 HAL?

This might introduce a timer conflict for `mks_robin_stm32`, but I'm not too worried about that. If that happens a little extra work will be needed to possibly move the TONE_TIMER, but that would be the right thing to do.

I verified that PC13 isn't attached to any timer pins. The Maple framework will take over any timer attached to BEEPER_PIN if SPEAKER is enabled, which can make enabling it risky for some boards. It seems ok for this board.

Reply:
The beeper works fine when compiled with mks_robin on my board, i should try compiling with mks_robin_stm32 to see if everything is still fine but for now i have other issues to solve to make marlin compile on this framework with my config...

Reply:
>I wonder why this was -1 when using the STM32 HAL?

There was a bug that caused Marlin with HAL STM32 hangs when speaker enabled.
I have not tested this for a while, so I can't tell if this bug still present.

Reply:
If you tested before I upgraded to ststm32 8.0 that might be fixed. I think I remember TONE not working on one timers, but it worked with the newer framework.

Reply:
The only problem in BEEPER_PIN that I know is when SPEAKER is enabled. I'm using the BEEPER in a STM32F4, on HAL STM32 with no problem, in the current version.